gui/default/syncthing/core/syncthingController.js

```diff
@@ -563,9 +578,53 @@ angular.module('syncthing.core')
                 }
                 $scope.connections = data;
                 console.log("refreshConnections", data);
+
+                refreshRemoteGUI(data);
             }).error($scope.emitHTTPError);
         }
 
+        function refreshRemoteGUI(connections) {
+            if (!$scope.showRemoteGUI) {
+                $scope.idToRemoteGUI = {}
+                return
+            }
+            var newCache = {};
+            for (var id in connections) {
+                if (!connections.hasOwnProperty(id)) {
+                    continue;
+                }
+                if (!(id in $scope.devices)) {
+                    // Avoid errors when called before first updateLocalConfig()
+                    continue;
+                }
+                if (!(id in $scope.idToRemoteGUI)) {
+                    $scope.idToRemoteGUI[id] = "";
+                }
+                var port = $scope.devices[id].remoteGUIPort;
+                var isNotRelayConnection = !connections[id].type.includes("relay");
+                if (connections[id].address != "" && isNotRelayConnection && port > 0) {
```

Comment:
I'd check and abort on `port == 0` before all of this to avoid having potentially lots of useless empty string entries in `idToRemoteGUI` (potentially deleting the entry for `id` in case it was `port > 0` before).

Reply:
The check used to be explicitly for `idToRemoteGUI[id] == ""`, hence all the empty values.  I changed that later for simplicitly, so you're right this can now be avoided.

Reply:
> I'd check and abort on `port == 0` before all of this to avoid having potentially lots of useless empty string entries in `idToRemoteGUI`

Hope you're okay with the new version, it at least tries to avoid needless operations when some prerequisite failed.  Namely, if the `port > 0` check fails, the `||` operator will short-circuit and not even check the connection type.